File: openapi/code_samples/csharp/api_locations_{postalCode}_getCityDetailsByPostalCode.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;
using System.Collections.Generic;

[ApiController]
[Route("api/[controller]")]
public class LocationController : ControllerBase
{
    [HttpGet]
    [Route("findByPostalCode/{postalCode}")]
    [SwaggerOperation(Summary = "Retrieves the city details by postal code.")]
    [ProducesResponseType(typeof(List<CityModel>), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult FindByPostalCode(string postalCode)
    {
        // Your implementation here to retrieve city details by postal code
        // Simulated data for demonstration purposes
        var cities = new List<CityModel>
        {
            new CityModel { Name = "City 1", PostalCode = postalCode },
            new CityModel { Name = "City 2", PostalCode = postalCode },
            new CityModel { Name = "City 3", PostalCode = postalCode }
        };

        return Ok(cities);
    }
}

public class CityModel
{
    public string Name { get; set; }
    public string PostalCode { get; set; }
}

File: openapi/code_samples/java/api_locations_{postalCode}_getCityDetailsByPostalCode.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.Arrays;
import java.util.List;

@RestController
@RequestMapping("/api/location")
public class LocationController {

    @GetMapping("/findByPostalCode/{postalCode}")
    public ResponseEntity<List<CityModel>> findByPostalCode(@PathVariable String postalCode) {
        // Your implementation here to retrieve city details by postal code
        // Simulated data for demonstration purposes
        List<CityModel> cities = Arrays.asList(
                new CityModel("City 1", postalCode),
                new CityModel("City 2", postalCode),
                new CityModel("City 3", postalCode)
        );

        return ResponseEntity.ok(cities);
    }
}

@Data
@AllArgsConstructor
class CityModel {
    private String name;
    private String postalCode;
}
